plugins/Hosting/js/reducers/hosting.js

```diff
@@ -0,0 +1,96 @@
+import { Map, List } from 'immutable'
+import * as constants from '../constants/constants.js'
+
+const initialState = Map({
+	numContracts: 0,
+	storage: 0,
+	usersettings: List([
+		Map({ name: 'Max Duration (Weeks)', value: 0, min: 12 }),
+		Map({ name: 'Collateral per TB per Month (SC)', value: 0 }),
+		Map({ name: 'Price per TB per Month (SC)', value: 0 }),
+		Map({ name: 'Bandwidth Price (SC/TB)', value: 0 }),
+	]),
+	defaultsettings: undefined,
+	files: List([]),
+	earned: 0,
+	expected: 0,
+	acceptingContracts: 0,
+	settingsChanged: false,
+	walletLocked: true,
+	defaultAnnounceAddress: '',
+	modals: Map({
+		shouldShowResizeDialog: false,
+		resizePath: '',
+		resizeSize: 0,
+		initialSize: 0,
+		warningModalTitle: '',
+		warningModalMessage: '',
+		shouldShowWarningModal: false,
+		shouldShowAnnounceDialog: false,
+		announceAddress: '',
+	}),
+})
+
+export default function hostingReducer(state = initialState, action) {
+	switch (action.type) {
+	case constants.UPDATE_SETTING:
+		let settingslist = state.get('usersettings')
+		settingslist = settingslist.map((value) => (
+			(value.get('name') === action.setting) ?
+				value.set('value', action.value)
+			: value
+		))
```

Comment:
I definitely think settings should be held in their own reducer and have their own actions, passing around a big chunk of state with `updateSettings` and then mapping over that chunk seems like we're just implementing a _worse_ reducer that's harder to reason about.